setup.py

```diff
@@ -66,10 +66,10 @@ def parse_readme(readme: str) -> str:
     "fastapi",
     "fsspec<=2023.5.0",
     "pyarrow",
-    "pyOpenSSL>=21.1.0",
+    "pyOpenSSL>=23.3.0",
```

Comment:
curious was the previous version of pyopenssl breaking something in kubernetes? also, if you could also update the version in `requirements.txt` to match this